identity/src/ed25519.rs

```diff
@@ -184,8 +184,11 @@ impl SecretKey {
     /// Generate a new Ed25519 secret key.
     #[cfg(feature = "rand")]
     pub fn generate() -> SecretKey {
-        let signing = ed25519::SigningKey::generate(&mut rand::rngs::OsRng);
-        SecretKey(signing.to_bytes())
+        use rand::RngCore as _;
+
+        let mut secret = [0; ed25519::SECRET_KEY_LENGTH];
```

Comment:
nit: wdyt of just `default`'ing as the internal `ed25519` does? 
```suggestion
        let mut secret = ed25519::SecretKey::default();
```